tests/unit/lms/views/api/blackboard/files_test.py

```diff
@@ -19,14 +24,99 @@ def test_it(
         files = view()
 
         blackboard_api_client.request.assert_called_once_with(
-            "GET", "courses/uuid:COURSE_ID/resources"
+            "GET", "courses/uuid:COURSE_ID/resources?limit=200"
         )
         BlackboardListFilesSchema.assert_called_once_with(
             blackboard_api_client.request.return_value
         )
         assert files == blackboard_list_files_schema.parse.return_value
 
-    @pytest.fixture
+    def test_it_with_pagination(
+        self, view, blackboard_api_client, blackboard_list_files_schema
+    ):
+        blackboard_api_client.request.side_effect = [
+            # The first response from the Blackboard API includes the path to
+            # the next page in the JSON body.
+            factories.requests.Response(
+                json_data={
+                    "paging": {
+                        # The Blackboard API gives us the whole path to the next
+                        # page, including limit and offset query params, as a
+                        # string. For example:
+                        # "/learn/api/public/v1/courses/uuid:<ID>/resources?limit=200&offset=200"
+                        "nextPage": "PAGE_2_PATH"
+                    }
+                }
+            ),
+            # The second response contains the path to the third page.
+            factories.requests.Response(
+                json_data={"paging": {"nextPage": "PAGE_3_PATH"}}
+            ),
+            # The third response is the last page: no path to the next page.
+            factories.requests.Response(json_data={}),
+        ]
+        blackboard_list_files_schema.parse.side_effect = [
+            # The first response from the Blackboard API contains the first
+            # three files.
+            [sentinel.file_1, sentinel.file_2, sentinel.file_3],
+            # The second page's files are in the second response.
+            [sentinel.file_4, sentinel.file_5, sentinel.file_6],
+            # The final page of files.
+            [sentinel.file_7, sentinel.file_8],
+        ]
```

Comment:
The way this is laid out makes it a bit hard for me to work out what is being setup here. I think in combination with reducing the number of return values we can slim this down:

```suggestion
        # The first response from the Blackboard API includes the path to
        # the next page in the JSON body.

        # The Blackboard API gives us the whole path to the next
        # page, including limit and offset query params, as a
        # string. For example:
        # "/learn/api/public/v1/courses/uuid:<ID>/resources?limit=200&offset=200"

        blackboard_api_client.request.side_effect = [
            factories.requests.Response(json_data=data)
            for data in [
                {"paging": {"nextPage": "PAGE_2_PATH"}},
                {"paging": {"nextPage": "PAGE_3_PATH"}},
                {},
            ]
        ]

        # Multiple sets of values which are returned one after another
        blackboard_list_files_schema.parse.side_effect = [
            [sentinel.file_1],
            [sentinel.file_2],
            [sentinel.file_3],
        ]
```

Reply:
:+1: Done